### PR TITLE
Add meganav to About page

### PR DESCRIPTION
**Pull request by azangru**
## Description
This PR adds a mega-nav panel to the About page, following the design in [this XD](https://xd.adobe.com/view/0cbbd0dc-10dc-461f-b62f-ba2c2b5a9d97-59e8/screen/f77789db-ef31-4e70-a300-4c113ce80efc/). The purpose of this PR is to support the restructuring of the articles for the About us section that is currently happening in [this branch](https://github.com/Ensembl/ensembl-website-help-and-docs/tree/feature/about-us) of ensembl-website-help-and-docs repo.

This PR will be ready when the articles ensembl-website-help-and-docs repo have been moved to appropriate locations.

**Note** that this PR does not disable the active link in the mega-nav, and does not add the chevrons to the top-level links, as seen in the XD. These changes would be common for both the Help app and the About app, and will be done in a separate PR.

## Related JIRA Issue(s)
https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-1397

## Deployment URL(s)
http://about-review.review.ensembl.org/

### CODE REVIEW

File: src/content/app/about/components/about-menu/AboutMenu.tsx [hunk 1]
```diff
@@ -48,33 +51,53 @@ export const TopMenu = (props: Props) => {
 export const SideMenu = (props: Props) => {
```
Comment (jyothishnt) [inline src/content/app/about/components/about-menu/AboutMenu.tsx:51]: Can we remove this export as this is no more used outside this component?

File: src/server/middleware/proxyMiddleware.ts [hunk 2]
```diff
@@ -56,13 +56,22 @@ const staticAssetsMiddleware = createProxyMiddleware('/static', {
...
-const apiProxyMiddleware = createProxyMiddleware('/api', {
+const apiProxyMiddleware = createProxyMiddleware(['/api/**', '!/api/docs/**'], {
   target: 'https://staging-2020.ensembl.org',
   changeOrigin: true,
   secure: false
 });
 
-let proxyMiddleware = [apiProxyMiddleware];
+const docsProxyMiddleware = createProxyMiddleware('/api/docs/**', {
+  target: 'http://localhost:3000',
```
Comment (jyothishnt) [inline src/server/middleware/proxyMiddleware.ts:66]: was this added to test locally?  we have api/docs/** mentioned in the above apoProxyMiddleware.

File: src/content/app/about/components/about-menu/AboutMenu.tsx [hunk 3]
```diff
@@ -48,33 +51,53 @@ export const TopMenu = (props: Props) => {
...
+          currentUrl,
+          index
+        )
+      )}
     </>
   );
 };
 
+const buildSideMenuLink = (
+  menuItem: { name: string; url: string },
```
Comment (imransl) [inline src/content/app/about/components/about-menu/AboutMenu.tsx:85]: Isn't `Pick<MenuCollectionItem, 'name' | 'url'>` better here?

File: src/server/middleware/proxyMiddleware.ts (hunk 2 above)
Comment (azangru) [reply src/server/middleware/proxyMiddleware.ts:66]: Yes, thank you! This shouldn't have gone in the PR.

File: src/content/app/about/components/about-menu/AboutMenu.tsx (hunk 1 above)
Comment (azangru) [reply src/content/app/about/components/about-menu/AboutMenu.tsx:51]: I'm removing the whole AboutMenu component. It isn't being used any longer. Thanks for pointing this out.

File: src/content/app/about/components/about-menu/AboutMenu.tsx (hunk 3 above)
Comment (azangru) [reply src/content/app/about/components/about-menu/AboutMenu.tsx:85]: The whole file is gone now :-)

**Review by jyothishnt — APPROVED** [review]

**Review by imransl — APPROVED** [review]

**Pull request merged**